### PR TITLE
Use Dagger's AndroidInjectionModule.

### PR DESCRIPTION
**Pull request by NightlyNexus**

### CODE REVIEW

File: telecine/build.gradle [hunk 1]
```diff
@@ -6,7 +6,7 @@ def versionBuild = 0 // bump for dogfood builds, public betas, etc.
 
 buildscript {
   dependencies {
-    classpath 'com.android.tools.build:gradle:3.0.0-alpha3'
+    classpath 'com.android.tools.build:gradle:3.0.0-alpha8'
```
Comment (NightlyNexus) [inline telecine/build.gradle]: oops, can be alpha9.

File: telecine/src/main/java/com/jakewharton/telecine/TelecineComponent.java [hunk 2]
```diff
@@ -1,15 +1,18 @@
...
 
+import android.app.Application;
+import dagger.BindsInstance;
 import dagger.Component;
+import dagger.android.AndroidInjectionModule;
 import javax.inject.Singleton;
 
-@Singleton
-@Component(modules = TelecineModule.class)
+@Singleton @Component(modules = { AndroidInjectionModule.class, TelecineModule.class })
```
Comment (NightlyNexus) [inline telecine/src/main/java/com/jakewharton/telecine/TelecineComponent.java]: do you happen to know why this seemed to work fine without declaring `AndroidInjectionModule.class` in the modules?

File: telecine/build.gradle [hunk 3]
```diff
@@ -78,13 +78,15 @@ dependencies {
   implementation 'com.android.support:support-annotations:25.1.0'
   implementation 'com.android.support:design:25.1.0'
   implementation 'com.google.android.gms:play-services-analytics:10.0.1'
-  implementation 'com.nightlynexus.demomode:demomode:0.1.1'
+  implementation 'com.nightlynexus.demomode:demomode:0.1.2'
```
Comment (JakeWharton) [inline telecine/build.gradle]: I would prefer this in a separate PR or at least a separate commit.

Comment (NightlyNexus) [inline telecine/build.gradle]: yep, 👍

File: telecine/src/main/java/com/jakewharton/telecine/TelecineModule.java [hunk 4]
```diff
@@ -44,20 +38,21 @@
     return new Analytics.GoogleAnalytics(tracker);
   }
 
-  @Provides @Singleton ContentResolver provideContentResolver() {
+  @Provides @Singleton static ContentResolver provideContentResolver(Application app) {
```
Comment (vanniktech) [inline telecine/src/main/java/com/jakewharton/telecine/TelecineModule.java]: the application parameter is not needed

File: telecine/src/main/java/com/jakewharton/telecine/TelecineModule.java [hunk 5]
```diff
@@ -44,20 +38,21 @@
...
     return app.getContentResolver();
   }
 
-  @Provides @Singleton SharedPreferences provideSharedPreferences() {
+  @Provides @Singleton static SharedPreferences provideSharedPreferences(Application app) {
```
Comment (vanniktech) [inline telecine/src/main/java/com/jakewharton/telecine/TelecineModule.java]: the application parameter is not needed

File: telecine/src/main/java/com/jakewharton/telecine/TelecineModule.java [hunk 6]
```diff
@@ -23,13 +23,7 @@
...
   private static final int DEFAULT_VIDEO_SIZE_PERCENTAGE = 100;
 
-  private final TelecineApplication app;
-
-  TelecineModule(TelecineApplication app) {
-    this.app = app;
-  }
-
-  @Provides @Singleton Analytics provideAnalytics() {
+  @Provides @Singleton static Analytics provideAnalytics(Application app) {
```
Comment (vanniktech) [inline telecine/src/main/java/com/jakewharton/telecine/TelecineModule.java]: the application parameter is not needed

File: telecine/src/main/java/com/jakewharton/telecine/TelecineService.java [hunk 7]
```diff
@@ -122,6 +123,11 @@ static Intent newIntent(Context context, int resultCode, Intent data) {
     }
   };
 
+  @Override public void onCreate() {
```
Comment (vanniktech) [inline telecine/src/main/java/com/jakewharton/telecine/TelecineService.java]: Could be a `DaggerService` and then this would not be needed

File: telecine/src/main/java/com/jakewharton/telecine/TelecineModule.java (hunk 4 above)
Comment (NightlyNexus) [inline telecine/src/main/java/com/jakewharton/telecine/TelecineModule.java]: oh, what am i missing? what would this method look like without the parameter?

File: telecine/src/main/java/com/jakewharton/telecine/TelecineService.java (hunk 7 above)
Comment (NightlyNexus) [inline telecine/src/main/java/com/jakewharton/telecine/TelecineService.java]: yeah, true, but it seems weird to have one place that extends the Dagger library types to hide the code and then the rest of the places have the code themselves.
